src/app/game-state/impossibleTask.service.ts

```diff
@@ -169,12 +169,29 @@ export class ImpossibleTaskService {
     });
   }
 
+  markPriorCompletions(){
+    for (let taskIndex = this.tasks.length - 1; taskIndex >= 0; taskIndex--){
```

Comment:
To my knowledge, `--` is ~30% slower in the V8 engine. Please ensure the tradeoff is worthwhile.

Reply:
After some testing, this knowledge might be dated.